src/main/java/com/aceleradora/conversor/Moeda.java

```diff
@@ -0,0 +1,42 @@
+package com.aceleradora.conversor;
+
+import java.math.BigDecimal;
+import java.util.stream.Stream;
+
+public enum Moeda {
+
+    DOLAR_AMERICANO("USD", new BigDecimal("1")),
```

Comment:
Nesta solução, cada moeda possui uma taxa de conversão para o dólar. Desta maneira, podemos usar o dólar como moeda intermediária de conversão e não necessitamos armazenar uma lista de taxas de câmbio de uma moeda para outra. 

Para usar o dólar como moeda intermediária, usamos a seguinte fórmula:

```
valor_saida = (moeda_original * taxa_dolar) / taxa_dolar_moeda_saida
```

Esta ideia é inspirada/roubada do [NUC (Neutral Unit of Construction)](https://en.wikipedia.org/wiki/Neutral_unit_of_construction), uma moeda utilizada na indústria aérea para efetuar conversões de valores. Um `NUC` equivale a um `USD`.